sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,15 +31,174 @@
+
+    Intel Scalable Dataframe Compiler Developer Guide
+    *************************************************
+    Pandas DataFrame method :meth:`pandas.DataFrame.append` implementation.
+    .. only:: developer
+    Test: python -m sdc.runtests -k sdc.tests.test_dataframe.TestDataFrame.test_append*
+    Parameters
+    -----------
+    df: :obj:`pandas.DataFrame`
+        input arg
+    other: :obj:`pandas.DataFrame` object or :obj:`pandas.Series` or :obj:`dict`
+        The data to append
+    ignore_index: :obj:`bool`
+        *unsupported*
+    verify_integrity: :obj:`bool`
+        *unsupported*
+    sort: :obj:`bool`
+        *unsupported*
+
+    Returns
+    -------
+    :obj: `pandas.DataFrame`
+        return DataFrame with appended rows to the end
+    """
+
+    _func_name = 'append'
+
+    ty_checker = TypeChecker(f'Method {_func_name}().')
+    ty_checker.check(df, DataFrameType)
+    # TODO: support other array-like types
+    ty_checker.check(other, DataFrameType)
+    # TODO: support index in series from df-columns
+    if not isinstance(ignore_index, (bool, types.Boolean, types.Omitted)) and not ignore_index:
+        ty_checker.raise_exc(ignore_index, 'boolean', 'ignore_index')
+
+    if not isinstance(verify_integrity, (bool, types.Boolean, types.Omitted)) and verify_integrity:
+        ty_checker.raise_exc(verify_integrity, 'boolean', 'verify_integrity')
+
+    if not isinstance(sort, (bool, types.Boolean, types.Omitted)) and sort is not None:
+        ty_checker.raise_exc(sort, 'boolean, None', 'sort')
+
+    args = (('ignore_index', True), ('verify_integrity', False), ('sort', None))
+
+    def sdc_pandas_dataframe_append_impl(df, other, _func_name, args):
+        indent = 4 * ' '
+        func_args = ['df', 'other']
+
+        for key, value in args:
+            # TODO: improve check
+            if key not in func_args:
+                if isinstance(value, types.Literal):
+                    value = value.literal_value
+                func_args.append('{}={}'.format(key, value))
+
+        df_columns_indx = {col_name: i for i, col_name in enumerate(df.columns)}
+        other_columns_indx = {col_name: i for i, col_name in enumerate(other.columns)}
+
+        # Keep columns that are StringArrayType
+        string_type_columns = set(col_name for typ, col_name in zip(df.data, df.columns)
+                                  if isinstance(typ, StringArrayType))
+
+        for typ, col_name in zip(other.data, other.columns):
+            if isinstance(typ, StringArrayType):
+                string_type_columns.add(col_name)
+
+        def get_dataframe_column(df, column, idx):
+            return f'new_col_{column}_data_{df} = get_dataframe_data({df}, {idx})'
+
+        def get_append_result(df1, df2, column):
+            return f'new_col_{column} = ' \
+                f'init_series(new_col_{column}_data_{df1}).append(init_series(new_col_{column}_data_{df2}))._data'
```

Comment:
I would create series strings separately then insert in to the resulting string, something like that:
```
s1 = f'init_series(new_col_{column}_data_{df1})'
s2 = f'init_series(new_col_{column}_data_{df2})'
return f'new_col_{column} = {s1}.append{s2}._data
```

Reply:
It can help to avoid long code lines.